pkg/provider/gcp/secretmanager/client.go

```diff
@@ -497,10 +499,21 @@ func (c *Client) GetSecret(ctx context.Context, ref esv1.ExternalSecretDataRemot
 		Name: name,
 	}
 	result, err := c.smClient.AccessSecretVersion(ctx, req)
-	metrics.ObserveAPICall(constants.ProviderGCPSM, constants.CallGCPSMAccessSecretVersion, err)
-	err = parseError(err)
 	if err != nil {
-		return nil, fmt.Errorf(errClientGetSecretAccess, err)
+		if c.store.GetLatestEnabledSecret && ref.Version == "" {
+			if isErrSecretDestroyedOrDisabled(err) {
+				result, err = getLatestEnabledVersion(ctx, c.smClient, name)
+				if err != nil {
+					metrics.ObserveAPICall(constants.ProviderGCPSM, constants.CallGCPSMAccessSecretVersion, err)
+					err = parseError(err)
+					return nil, fmt.Errorf(errClientGetSecretAccess, err)
+				}
+			}
+		} else {
+			metrics.ObserveAPICall(constants.ProviderGCPSM, constants.CallGCPSMAccessSecretVersion, err)
+			err = parseError(err)
+			return nil, fmt.Errorf(errClientGetSecretAccess, err)
+		}
 	}
```

Comment:
Can we flip this around to avoid the too many nested if statements please?

Something like, if there is no error, do the thing, then fall back to error handling.

Reply:
You're right. It's possible and I just refactored this :)

Reply:
Thank you! 🙇